stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/attachments/LinkAttachmentFactory.kt

```diff
@@ -0,0 +1,134 @@
+package io.getstream.chat.android.compose.ui.attachments
+
+import android.content.Intent
+import android.net.Uri
+import androidx.compose.foundation.ExperimentalFoundationApi
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.background
+import androidx.compose.foundation.combinedClickable
+import androidx.compose.foundation.interaction.MutableInteractionSource
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.heightIn
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.layout.wrapContentHeight
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.unit.dp
+import coil.compose.rememberImagePainter
+import com.getstream.sdk.chat.utils.extensions.imagePreviewUrl
+import io.getstream.chat.android.client.models.Message
+import io.getstream.chat.android.compose.state.messages.attachments.AttachmentState
+import io.getstream.chat.android.compose.ui.theme.ChatTheme
+
+/**
+ * An extension of the [AttachmentFactory] that validates attachments as images and uses [LinkAttachmentContent] to
+ * build the UI for the message.
+ * */
+public class LinkAttachmentFactory : AttachmentFactory() {
+
+    override fun predicate(message: Message): Boolean {
+        return message.attachments.any { it.titleLink != null || it.ogUrl != null }
+    }
+
+    @Composable
+    public override fun Content(attachmentState: AttachmentState) {
```

Comment:
Nitpick of nitpicks: visibility modifiers are redundant when overriding a `public` method (applies in other files too)

Reply:
Okay - it initially showed me an error as per the linting rules, will remove.